content/en/docs/guides/deadlines.md

```diff
@@ -0,0 +1,77 @@
+---
+title: Deadlines
+description: >-
+  Explains how deadlines can be used to effectively deal with unreliable
+  backends.
+---
+
+### Overview
+
+A deadline is used to specify how long a client is willing to wait for a server
+to respond. This simple idea is very important in building robust distributed
+systems. Clients that do not wait around unnecessarily and servers that know
+when to give up processing requests will improve the resource utilization and
+latency of your system.
+
+Note that while some language APIs have the concept of a __deadline__, others
+use the idea of a __timeout__. When an API asks for a deadline, you provide a
+point in time which the request should not go past. A timeout is the max
+duration of time that the request can take. For simplicity, we will only refer
+to deadline in this document.
+
+### Deadlines on the Client
+
+The default deadline for a gRPC client is some very large number that would not
+be appropriate to use in any real world scenario. This means you should always
+set a realistic deadline in your clients. To determine the appropriate deadline
+you would ideally start with an educated guess based on what you know about your
+system (network latency, server processing time, etc.), validated by some load
+testing.
+
+If a server has gone past the deadline when processing a request, the client
+will give up and fail the RPC with the `DEADLINE_EXCEEDED` status. Note that the
+client giving up on a particular RPC does not automatically mean the server will
+stop processing the request. The server needs to take precautions to prevent
```

Comment:
It would be very helpful to give a couple of examples of precautions that the server could take.

Reply:
The next section goes into that in more detail. I was rephrasing the sentence, but ended up just deleting it.